sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/jsonschema/JsonSchemaResource.java

```diff
@@ -0,0 +1,180 @@
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.docsapi.api.v2.namespaces.collections.jsonschema;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.smallrye.mutiny.Uni;
+import io.stargate.sgv2.docsapi.api.common.exception.model.dto.ApiError;
+import io.stargate.sgv2.docsapi.api.v2.namespaces.collections.model.dto.JsonSchemaDto;
+import io.stargate.sgv2.docsapi.config.constants.OpenApiConstants;
+import io.stargate.sgv2.docsapi.service.schema.JsonSchemaManager;
+import io.stargate.sgv2.docsapi.service.schema.TableManager;
+import io.stargate.sgv2.docsapi.service.schema.qualifier.Authorized;
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.UriInfo;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.ExampleObject;
+import org.eclipse.microprofile.openapi.annotations.media.SchemaProperty;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameters;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.jboss.resteasy.reactive.RestResponse;
+
+/** JSON Schema resource. */
+@Path(JsonSchemaResource.BASE_PATH)
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@SecurityRequirement(name = OpenApiConstants.SecuritySchemes.TOKEN)
+@Tag(ref = OpenApiConstants.Tags.JSON_SCHEMAS)
+public class JsonSchemaResource {
+
+  public static final String BASE_PATH =
+      "/v2/namespaces/{namespace:\\w+}/collections/{collection:\\w+}/json-schema";
+
+  @Inject @Authorized TableManager tableManager;
+
+  @Inject JsonSchemaManager jsonSchemaManager;
+
+  @Operation(
+      description =
+          "Assign a JSON schema to a collection. This will erase any schema that already exists.")
+  @Parameters(
+      value = {
+        @Parameter(
+            name = "namespace",
+            ref = OpenApiConstants.Parameters.NAMESPACE,
+            description = "The namespace of the collection"),
+        @Parameter(
+            name = "collection",
+            ref = OpenApiConstants.Parameters.COLLECTION,
+            description = "The collection to assign a JSON schema to.")
+      })
+  @APIResponses(
```

Comment:
should we also add an example for the body here.. I know it can be any valid JSON schema, but right now it looks wierd with empty body.. You would need a `@RequestBody` open api annotation for this..

![image](https://user-images.githubusercontent.com/10600041/172118725-ec49e970-e5d2-45bd-be6e-e9d630a8c461.png)